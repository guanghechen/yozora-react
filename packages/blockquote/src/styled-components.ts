import 'styled-components'
import type { YozoraBlockquoteTheme } from './theme'


declare module 'styled-components' {
  export interface DefaultTheme {
    yozora: YozoraTheme
  }

  interface YozoraTheme {
    blockquote?: YozoraBlockquoteTheme
  }
}
