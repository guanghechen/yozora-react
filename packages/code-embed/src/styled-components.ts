import type { YozoraCodeEmbedTheme } from './theme'
import 'styled-components'


declare module 'styled-components' {
  export interface DefaultTheme {
    yozora: YozoraTheme
  }

  interface YozoraTheme {
    codeEmbed?: YozoraCodeEmbedTheme
  }
}
