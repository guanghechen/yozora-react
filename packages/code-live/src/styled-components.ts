import 'styled-components'
import type { YozoraCodeLiveTheme } from './theme'


declare module 'styled-components' {
  export interface DefaultTheme {
    yozora: YozoraTheme
  }

  interface YozoraTheme {
    codeLive?: YozoraCodeLiveTheme
  }
}
