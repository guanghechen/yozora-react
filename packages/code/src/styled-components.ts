import 'styled-components'
import type { YozoraCodeTheme } from './theme'


declare module 'styled-components' {
  export interface DefaultTheme {
    yozora: YozoraTheme
  }

  interface YozoraTheme {
    code?: YozoraCodeTheme
  }
}
