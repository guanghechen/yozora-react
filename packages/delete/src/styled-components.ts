import type { YozoraDeleteTheme } from './theme'
import 'styled-components'


declare module 'styled-components' {
  export interface DefaultTheme {
    yozora: YozoraTheme
  }

  interface YozoraTheme {
    delete?: YozoraDeleteTheme
  }
}
