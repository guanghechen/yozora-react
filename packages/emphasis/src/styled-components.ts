import 'styled-components'
import type { YozoraEmphasisTheme } from './theme'


declare module 'styled-components' {
  export interface DefaultTheme {
    yozora: YozoraTheme
  }

  interface YozoraTheme {
    emphasis?: YozoraEmphasisTheme
  }
}
