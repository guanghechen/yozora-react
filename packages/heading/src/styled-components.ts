import 'styled-components'
import type { YozoraHeadingTheme } from './theme'


declare module 'styled-components' {
  export interface DefaultTheme {
    yozora: YozoraTheme
  }

  interface YozoraTheme {
    heading?: YozoraHeadingTheme
  }
}
