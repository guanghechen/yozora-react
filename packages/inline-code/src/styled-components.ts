import 'styled-components'
import type { YozoraInlineCodeTheme } from './theme'


declare module 'styled-components' {
  export interface DefaultTheme {
    yozora: YozoraTheme
  }

  interface YozoraTheme {
    inlineCode: YozoraInlineCodeTheme
  }
}
