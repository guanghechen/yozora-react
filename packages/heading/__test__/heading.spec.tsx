import React from 'react'
import { mount, render } from 'enzyme'
import { DefaultTheme, ThemeProvider } from 'styled-components'
import Heading from '../src'


describe('basic rendering case', () => {
  const errorLogger = jest
    .spyOn(global.console, 'error')
    .mockImplementation((...args) => {
      throw new Error(args.join(' '))
    })

  afterAll(() => {
    errorLogger.mockRestore()
  })

  it('render a simple content', () => {
    const text = 'Hello, world!'
    const wrapper = render(
      <Heading level={ 3 } identifier="heading-22">{ text }</Heading>
    )
    expect(wrapper.text()).toEqual(text)
  })

  it('render with custom className', () => {
    const text = 'Hello, world!'
    const className = 'custom-heading'
    const wrapper = render(
      <Heading level={ 1 } identifier="heading-33" className={ className }>
        <span>{ text }</span>
      </Heading>
    )
    expect(wrapper.hasClass(className)).toEqual(true)
    expect(wrapper.text()).toEqual(text)
  })

  it('level is a required enum number', () => {
    for (const level of [0, '1', 1.2, 7] as any[]) {
      expect(() => {
        render(
          <Heading level={ level } identifier={ `heading-${ level }` }>
            heading { level }
          </Heading>
        )
      }).toThrow(/Failed prop type: Invalid prop `level`/i)
    }
  })

  it('children is required', () => {
    for (const value of [undefined, null] as any[]) {
      expect(() => {
        render(
          <Heading level={ 1 } identifier="heading-1">
            { value }
          </Heading>
        )
      }).toThrow(/Failed prop type: The prop `children` is marked as required/i)
    }
  })

  it('forward ref', () => {
    const ref = React.createRef<HTMLDivElement>()
    const level = 1
    const wrapper = mount(
      <Heading level={ level } ref={ ref } identifier={ `heading-${ level }` } data-value="waw">
        Heading { level }
      </Heading>
    )

    const o = wrapper.getDOMNode()
    expect(o).toEqual(ref.current)
    expect(o.getAttribute('data-value')).toEqual('waw')
  })

  it('snapshot', () => {
    const level = 2
    const wrapper = render(
      <Heading
        level={ level }
        identifier={ `heading-${ level }` }
        style={ { color: 'orange', fontSize: '16px' } }
      >
        Waw -- { level }, 中文标题“这”
      </Heading>
    )
    expect(wrapper).toMatchSnapshot(`level ${ level }`)
  })

  it('snapshot with theme', () => {
    const theme: DefaultTheme = {
      yozora: {
        heading: {
          fontStyle: 'italic',
          color: '#ccc',
          padding: '0 2em',
          borderColor: 'lightgray',
          // margin: '1.2em -2em 1em',
          lineHeight: 1.25,
          fontFamily: 'inherit',
          h1FontSize: '2em',
          h2FontSize: '1.5em',
          h3FontSize: '1.25em',
          h4FontSize: '1em',
          h5FontSize: '0.875em',
          h6FontSize: '0.85em',
          linkColor: 'blue',
          hoverLinkColor: 'cyan',
        }
      }
    }

    const level = 1
    const wrapper = render(
      <ThemeProvider theme={ theme }>
        <Heading
          level={ level }
          identifier={ `heading-${ level }` }
          style={ { color: 'orange', fontSize: '16px' } }
        >
          Waw -- { level }, 中文标题“这”
        </Heading>
      </ThemeProvider>
    )
    expect(wrapper).toMatchSnapshot()
  })
})
