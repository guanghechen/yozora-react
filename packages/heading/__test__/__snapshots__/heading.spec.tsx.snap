// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`basic rendering case snapshot: level 2 1`] = `
.c0 {
  position: relative;
  display: block;
  margin: var(--md-heading-margin,1.2em -2em 1em);
  padding: var(--md-heading-padding,0 2em);
  line-height: var(--md-heading-line-height,1.25);
  font-family: var(--md-heading-font-family,inherit);
}

.c0 h1,
.c0 h2,
.c0 h3,
.c0 h4,
.c0 h5,
.c0 h6 {
  line-height: var(--md-heading-line-height,1.25);
  color: var(--md-heading-color,inherit);
  font-style: var(--md-heading-font-style,normal);
}

.c0 h1,
.c0 h2 {
  padding: 0.3em 0;
  border-bottom: 1px solid var(--md-heading-border-color);
  margin: 0.9em 0 1em;
}

.c0 h1 {
  font-size: var(--md-heading1-font-size,2em);
}

.c0 h2 {
  font-size: var(--md-heading2-font-size,1.5em);
}

.c0 h3 {
  font-size: var(--md-heading3-font-size,1.25em);
}

.c0 h4 {
  font-size: var(--md-heading4-font-size,1em);
}

.c0 h5 {
  font-size: var(--md-heading5-font-size,0.875em);
}

.c0 h6 {
  font-size: var(--md-heading6-font-size,0.85em);
}

.c0 > a {
  position: absolute;
  top: 50%;
  left: 0.25em;
  -webkit-transform: translate(0,-50%);
  -ms-transform: translate(0,-50%);
  transform: translate(0,-50%);
  display: inline-block;
  height: 1.25em;
  padding-right: 0.25em;
  font-size: 1.25em;
  line-height: 1;
  color: var(--md-heading-link-color,inherit);
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  visibility: hidden;
  opacity: 0;
}

.c0 > a > svg {
  overflow: hidden;
  display: inline-block;
  vertical-align: middle;
  fill: currentColor;
}

.c0:hover > a {
  color: var(--md-heading-link-hover-color,inherit);
  visibility: visible;
  opacity: 1;
}

<Heading
  level={2}
  style={
    Object {
      "color": "orange",
      "fontSize": "16px",
    }
  }
>
  <styled.header
    style={
      Object {
        "color": "orange",
        "fontSize": "16px",
      }
    }
  >
    <header
      className="c0"
      style={
        Object {
          "color": "orange",
          "fontSize": "16px",
        }
      }
    >
      <a
        href="#heading-waw-2-中文标题-这"
        id="heading-waw-2-中文标题-这"
      >
        <HeadingLinkIcon>
          <svg
            aria-hidden="true"
            height="16"
            version="1.1"
            viewBox="0 0 16 16"
            width="16"
          >
            <path
              d="M7.775 3.275a.75.75 0 001.06 1.06l1.25-1.25a2 2 0 112.83 2.83l-2.5 2.5a2 2 0 01-2.83 0 .75.75 0 00-1.06 1.06 3.5 3.5 0 004.95 0l2.5-2.5a3.5 3.5 0 00-4.95-4.95l-1.25 1.25zm-4.69 9.64a2 2 0 010-2.83l2.5-2.5a2 2 0 012.83 0 .75.75 0 001.06-1.06 3.5 3.5 0 00-4.95 0l-2.5 2.5a3.5 3.5 0 004.95 4.95l1.25-1.25a.75.75 0 00-1.06-1.06l-1.25 1.25a2 2 0 01-2.83 0z"
              fillRule="evenodd"
            />
          </svg>
        </HeadingLinkIcon>
      </a>
      <h2>
        Waw -- 
        2
        , 中文标题“这”
      </h2>
    </header>
  </styled.header>
</Heading>
`;
